slides/solidjs.md

```diff
@@ -1,7 +1,590 @@
+- [useEffect](https://reactjs.org/docs/hooks-reference.html#useeffect) <mdi-arrow-right /> [createEffect](https://www.solidjs.com/docs/latest/api#createeffect)
+- [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) <mdi-arrow-right /> [createMemo](https://www.solidjs.com/docs/latest/api#creatememo)
+- [useContext](https://reactjs.org/docs/hooks-reference.html#usecontext) <mdi-arrow-right /> [useContext](https://www.solidjs.com/docs/latest/api#usecontext)
+- [&lt;Suspense /&gt;](https://reactjs.org/docs/react-api.html#reactsuspense) <mdi-arrow-right /> [&lt;Suspense /&gt;](https://www.solidjs.com/docs/latest/api#suspense)
+- [useQuery/useSwr](https://swr.vercel.app/) <mdi-arrow-right /> [createResource](https://www.solidjs.com/docs/latest/api#createresource)
+- [and much more...](https://www.solidjs.com/docs/latest/api)
+
+</v-clicks>
+
+</div>
+
+---
+layout: center
+---
+
+# Solid component
+
+```typescript
+function Counter(props) {
+  const [count, setCount] = createSignal(props.initialCount ?? 0);
+
+  return (
+    <button 
+      type="button" 
+      onClick={() => setCount((n) => ++n)}
+    >
+      {count}
+    </button>
+  );
+}
+```
+
+A standard Solid component it's just a function that:
+- returns a DOM [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element)
+- receives props from the parent component
+- runs **once** per instance, in other words, does not rerender
+
+---
+
+<style>
+  .react-recon-pos {
+    right: 3rem;
+    top: 12rem;
+  }
+  .react-recon-count {
+    right: 20rem;
+    top: 10.2rem;
+    font-size: 1.4rem;
+  }
+</style>
+
+<h1 class="absolute">React reconciliation</h1>
+
+<div class="relative" style="padding: 2.5rem 1rem 1rem">
+  <div class="absolute react-recon-count">
+    count: <s>7</s>=><strong>8</strong>
+  </div>
+  <pre><code class="absolute react-recon-pos">setCount(n => n + 1);</code></pre>
+ <img src="/assets/react-reconciliation.svg" alt="react reconciliation" />
+</div>
+
+---
+
+<style>
+  .react-recon-pos {
+    right: 3rem;
+    top: 12rem;
+  }
+
+  .react-recon-count {
+    right: 20rem;
+    top: 10.2rem;
+    font-size: 1.4rem;
+  }
+</style>
+
+<h1 class="absolute">SolidJS update</h1>
+<div class="relative" style="padding: 2.5rem 1rem 1rem">
+  <div class="absolute react-recon-count">
+    count: <s>7</s>=><strong>8</strong>
+  </div>
+  <pre><code class="absolute react-recon-pos">setCount(n => n + 1);</code></pre>
+ <img src="/assets/solidjs-update.svg" alt="react reconciliation" />
+</div>
+
+---
+
+# Fine-Grained reactivity
+
+```typescript
+function createSignal(value) {
+  const subscribers = new Set();
+
+  const read = () => {
+    const listener = getCurrentListener(); // component, createEffect etc...
+    if (listener) subscribers.add(listener);
+    return value;
+  };
+
+  const write = (nextValue) => {
+    value = nextValue;
+    for (const sub of subscribers) sub.run();
+  };
+
+  return [read, write];
+}
+```
```

Comment:
Why are you showing this snippet here?

Reply:
I've rewritten the reactivity slides.

see
- https://deploy-preview-1--next-wave-fw-slides.netlify.app/14 
- https://deploy-preview-1--next-wave-fw-slides.netlify.app/15